src/mp4.c

```diff
@@ -678,27 +678,29 @@ _mp4_read_box(mp4info *mp4)
   type[4] = '\0';
   buffer_consume(mp4->buf, 4);
 
+  mp4->hsize = 8;
   // Check for 64-bit size
   if (size == 1) {
+    if ( !_check_buf(mp4->infile, mp4->buf, 8, MP4_BLOCK_SIZE) ) {
+      return 0;
+    }
     size = buffer_get_int64(mp4->buf);
     mp4->hsize = 16;
   }
-  else if (size == 0) {
-    // XXX: size extends to end of file
-    mp4->hsize = 8;
-  }
-  else {
-    mp4->hsize = 8;
-  }
-
-  if (size) {
+  if (size != 0) { // if (size == 0) // XXX: size extends to end of file
```

Comment:
I find the comment here a bit confusing. At first I thought it was left-over, commented out code.

Maybe write like this:
```
if (size == 0) {
  // XXX: size extends to end of file
}
else {
  // set size to the remainder of the box
  mp4->rsize = size - mp4->hsize;
}
```
Also, maybe abort parsing if "size < mp4->hsize" otherwise the uint8_t mp4->rsize will wrap around when it goes negative on line 692.

Feel free to ignore my comment if you don't agree. I'm not a maintainer, just someone dropping by, taking a look :-) I haven't worked with this code before..

Reply:
Good catch about the possibility of invalid header size data in the file. Added a check for that.

Your suggestion for the comment was my first idea too, but a lot of compilers will give a warning with the empty code block. I went ahead and did it that way with an empty command (just the ;). Thanks.